x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -176,7 +168,7 @@ public void stopDeployment(TrainedModelDeploymentTask task) {
     public void infer(TrainedModelDeploymentTask task,
                       String input, TimeValue timeout,
                       ActionListener<InferenceResults> listener) {
-        ProcessContext processContext = processContextByAllocation.get(task.getAllocationId());
+        ProcessContext processContext = processContextByAllocation.get(task.getId());
```

Comment:
task IDs are monotonically increasing for the life time of the TaskManager, so this is safe.